core/src/main/java/com/rudderstack/android/sdk/core/EventRepository.java

```diff
@@ -470,10 +470,21 @@ void dump(@NonNull RudderMessage message) {
         }
 
         RudderLogger.logDebug(String.format(Locale.US, "EventRepository: dump: eventName: %s", message.getEventName()));
-        Map<String, Object> integrations = new HashMap<>();
-        integrations.put("All", true);
-        message.setIntegrations(integrations);
-
+        // if no integrations were set in the RudderOption object passed in that particular event
+        // we would fall back to check for the integrations in the RudderOption object passed while initializing the sdk
+        if (message.getIntegrations().size() == 0) {
+            if(RudderClient.getDefaultOptions()!=null && RudderClient.getDefaultOptions().getIntegrations()!=null && RudderClient.getDefaultOptions().getIntegrations().size()!=0)
+            {
+                message.setIntegrations(RudderClient.getDefaultOptions().getIntegrations());
+            }
+            // if no RudderOption object is passed while initializing the sdk we would set all the integrations to true
+            else
+            {
+                Map<String, Object> integrations = new HashMap<>();
+                integrations.put("All", true);
+                message.setIntegrations(integrations);
```

Comment:
we can use the prepareIntegrations method instead directly?

Reply:
Yes